controllers/openstackcluster_controller.go

```diff
@@ -474,15 +477,17 @@ func bastionHashHasChanged(computeHash string, clusterAnnotations map[string]str
 	return latestHash != computeHash
 }
 
-func reconcileNetworkComponents(scope scope.Scope, cluster *clusterv1.Cluster, openStackCluster *infrav1.OpenStackCluster) error {
+func reconcileNetworkComponents(ctx context.Context, scope scope.Scope, cluster *clusterv1.Cluster, openStackCluster *infrav1.OpenStackCluster) error {
+	log := ctrl.LoggerFrom(ctx)
+
 	clusterName := fmt.Sprintf("%s-%s", cluster.Namespace, cluster.Name)
 
 	networkingService, err := networking.NewService(scope)
 	if err != nil {
 		return err
 	}
 
-	scope.Logger().Info("Reconciling network components")
+	log.Info("Reconciling network components")
```

Comment:
This seems to have worked:

* without that patch:
```
"Reconciling network components" controller="openstackcluster" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="OpenStackCluster" OpenStackCluster="e2e-b35o3t/cluster-e2e-b35o3t" namespace="e2e-b35o3t" name="cluster-e2e-b35o3t" reconcileID="1b3f9997-07dc-43e2-8e9b-f5d639bd4390" cluster="cluster-e2e-b35o3t"
```
count of reconcileID: 1000+

* with that patch:
```
"Reconciling network components" controller="openstackcluster" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="OpenStackCluster" OpenStackCluster="e2e-gik87x/cluster-e2e-gik87x" namespace="e2e-gik87x" name="cluster-e2e-gik87x" reconcileID="b35030d5-7612-46af-aac2-7473b5e35041"
```
count of reconcileID: 5

@mdbooth just confirming with you that my solution is ok with you and I'll replace all `scope.Logger()` by a `log` from `ctx`.

Thanks

Reply:
This looks great!